### PR TITLE
Basic test system to work with docker image

### PR DESCRIPTION
**Pull request by stillson**
This creates a basic system for working with the docker image rivos/rivos-qemu-docker which is what we use for acceptance testing with github actions.

Macros added to declare test pass and fail which will be detected when run under the overwatch.py program on the docker image.

The idea is that this can be easily extended to other parts of the system which can println!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -354,7 +359,10 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         // unless we have support for per-mode filtering.
         println!("Sscofpmf support present");
         if let Err(e) = PmuInfo::init() {
+            test_fail!("sstc presence");
             println!("PmuInfo::init() failed with {:?}", e);
+        } else {
+            test_pass!("sstc presence")
         }
     }
```
[the remark this answers is not in the document]
Comment (stillson) [reply src/main.rs:367]: Added test_declare_pass!() and test_runtest!()
test_declare_pass!() declares and passes in one macro

test_runtest!() executes a block. There is also test_assert!()

[the remark this answers is not in the document]
Comment (abrestic-rivos) [reply src/main.rs:366]: Can we convert the remaining instances of separate `test_declare!` + `test_pass!` pairs to a single invocation? Having both strings matches is brittle and I'd like to avoid checking it in...

File: src/main.rs [hunk 2]
```diff
@@ -354,7 +357,10 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         // unless we have support for per-mode filtering.
         println!("Sscofpmf support present");
         if let Err(e) = PmuInfo::init() {
+            test_declare_fail!("sstc presence");
```
Comment (abrestic-rivos) [inline src/main.rs:360]: s/sstc/sscofpmf

File: test-workloads/src/bin/guestvm.rs [hunk 3]
```diff
@@ -169,6 +171,8 @@ pub fn test_vector() {
         }
     }
 
+    test_assert!(!should_panic, "writing/reading vector registers");
+
```
Comment (abrestic-rivos) [inline test-workloads/src/bin/guestvm.rs:175]: This and the cert stuff below seems like good candidates for `test_runtest!`.

File: u-mode/src/main.rs [hunk 4]
```diff
@@ -121,5 +122,6 @@ extern "C" fn task_main(cpuid: u64, shared_addr: u64, shared_size: u64) -> ! {
         "umode/#{}: U-mode Shared Region: {:016x} - {} bytes",
         cpuid, shared_addr, shared_size
     );
+    test_declare_pass!("umode start", cpuid);
```
Comment (abrestic-rivos) [inline u-mode/src/main.rs:125]: minor: I think it would be better to do this in Salus after the U-mode task calls back into Salus for its next task in `run_loop()`.

Comment (stillson) [reply u-mode/src/main.rs:125]: or both.

File: src/main.rs (hunk 2 above)
Comment (stillson) [reply src/main.rs:360]: fixed

File: test-workloads/src/bin/guestvm.rs (hunk 3 above)
Comment (stillson) [reply test-workloads/src/bin/guestvm.rs:175]: These seem a bit to complicated to drop into a single block.

Comment (abrestic-rivos) [reply test-workloads/src/bin/guestvm.rs:175]: Seems pretty straightforward to me. Just make `test_vector` / `test_attestation` return `Pass` / `Fail` instead of panicing on error and then: `test_runtest!("...", || test_foo())`.

Comment (stillson) [reply test-workloads/src/bin/guestvm.rs]: fixed

File: src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stillson) [reply src/main.rs:365]: fixed

**Comment by abrestic-rivos** [comment]
Not sure what's going on with your git workflow, but this branch just looks like commits from main.

**Comment by stillson** [comment]
You caught it in the middle of a rebase. Trying to fix it now.

File: src/main.rs [hunk 5]
```diff
@@ -22,7 +22,7 @@
 
 use core::alloc::{Allocator, GlobalAlloc, Layout};
 use core::ptr::NonNull;
-
+use test_system::{test_declare, test_declare_fail, test_declare_pass, test_fail, test_pass};
```
Comment (abrestic-rivos) [inline src/main.rs:25]: I'd just do `use test_system::*`

File: src/main.rs [hunk 6]
```diff
@@ -345,6 +347,7 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         // We don't implement or use the SBI timer extension and thus require Sstc for timers.
         panic!("CPU does not support Sstc");
     }
+
```
Comment (abrestic-rivos) [inline src/main.rs:350]: unrelated whitespace change

**Review by abrestic-rivos — APPROVED** [review]

File: src/main.rs (hunk 5 above)
Comment (stillson) [reply src/main.rs]: fixed
